projects/pt1/python/torch_mlir_e2e_test/test_suite/pooling.py

```diff
@@ -1164,3 +1353,106 @@ def forward(self,x):
     module.forward(tu.rand(1, 512, 10, 16))
+
+# AdaptiveMaxPool3d
+
+class AdaptiveMaxPool3dDynamic(torch.nn.Module):
+    
+    def __init__(self):
+        super().__init__()
+        self.amp3d = torch.nn.AdaptiveMaxPool3d(output_size=(7,13,15), return_indices=False)
+
+    @export
+    @annotate_args([
+        None,
+        ([-1,-1,-1,-1,-1], torch.float32, True)
+    ])
+    def forward(self,x):
+        return self.amp3d(x)
+
+@register_test_case(
+    module_factory=lambda: AdaptiveMaxPool3dDynamic())
+def AdaptiveMaxPool3dDynamic_basic(
+        module, tu: TestUtils):
+    module.forward(tu.rand(1, 512, 10, 16, 17))
+
+class AdaptiveMaxPool3dDynamicNoBatch(torch.nn.Module):
+    
+    def __init__(self):
+        super().__init__()
+        self.amp3d = torch.nn.AdaptiveMaxPool3d(output_size=(7,13,15), return_indices=False)
+
+    @export
+    @annotate_args([
+        None,
+        ([-1,-1,-1,-1], torch.float32, True)
+    ])
+    def forward(self,x):
+        return self.amp3d(x)
+
+@register_test_case(
+    module_factory=lambda: AdaptiveMaxPool3dDynamicNoBatch())
+def AdaptiveMaxPool3dDynamicNoBatch_basic(
+        module, tu: TestUtils):
+    module.forward(tu.rand(512, 10, 16, 17))
+
+class AdaptiveMaxPool3dDynamicWithIndices(torch.nn.Module):
+    
+    def __init__(self):
+        super().__init__()
+        self.amp3d = torch.nn.AdaptiveMaxPool3d(output_size=(7,13,15), return_indices=True)
+
+    @export
+    @annotate_args([
+        None,
+        ([-1,-1,-1,-1,-1], torch.float32, True)
+    ])
+    def forward(self,x):
+        return self.amp3d(x)
+
+@register_test_case(
+    module_factory=lambda: AdaptiveMaxPool3dDynamicWithIndices())
+def AdaptiveMaxPool3dDynamicWithIndices_basic(
+        module, tu: TestUtils):
+    module.forward(tu.rand(1, 512, 10, 16, 17))
+    
+
+class AdaptiveMaxPool3dStatic(torch.nn.Module):
+    
+    def __init__(self):
+        super().__init__()
+        self.amp3d = torch.nn.AdaptiveMaxPool3d(output_size=(7,13,15), return_indices=False)
+
+    @export
+    @annotate_args([
+        None,
+        ([1, 512, 10, 9, 5], torch.float32, True)
+    ])
+    def forward(self,x):
+        return self.amp3d(x)
+
+@register_test_case(
+    module_factory=lambda: AdaptiveMaxPool3dStatic())
+def AdaptiveMaxPool3dStatic_basic(
+        module, tu: TestUtils):
+    module.forward(tu.rand(1, 512, 10, 9, 5))
+
+class AdaptiveMaxPool3dStaticWithIndices(torch.nn.Module):
+    
+    def __init__(self):
+        super().__init__()
+        self.amp3d = torch.nn.AdaptiveMaxPool3d(output_size=(7,13,15), return_indices=True)
+
+    @export
+    @annotate_args([
+        None,
+        ([1, 512, 10, 16, 17], torch.float32, True)
+    ])
+    def forward(self,x):
+        return self.amp3d(x)
+
+@register_test_case(
+    module_factory=lambda: AdaptiveMaxPool3dStaticWithIndices())
+def AdaptiveMaxPool3dStaticWithIndices_basic(
+        module, tu: TestUtils):
+    module.forward(tu.rand(1, 512, 10, 16, 17))
```

Comment:
Yeah, I tried to add enough tests to cover most of the non-trivial cases, but it would certainly be better to randomize some of the input/output dims. 

I'm not sure if that is what you meant by a numeric test beyond these e2e tests, which seem quite numeric to me since they compare the python output with the compiled-from-mlir output.

Reply:
oh cool! i didn't realize that the e2e tests did that. ty!